site/docs/hive.md

```diff
@@ -50,6 +50,19 @@ You should now be able to issue Hive SQL `SELECT` queries using the above table
 SELECT * from table_a;
 ```
 
+#### Using Hive Catalog
+Iceberg tables created using `HiveCatalog` are automatically registered with Hive. 
+
+##### Create an Iceberg table
+The first step is to create an Iceberg table using the Spark/Java/Python API and `HiveCatalog`. For the purposes of this documentation we will assume that the table is called `table_b` and that the table location is `s3://some_path/table_b`.
+TODO: what do we need to set up when we create this table programatically for everything to be registered correctly for read usage in Hive?
```

Comment:
@pvary What does one need to do in order to get the table set up properly for the Hive read path in this case? What I have tried to do so far is this, first create an Iceberg table using the HiveCatalog like so:
```
 PartitionSpec spec = PartitionSpec.unpartitioned();
 Schema schema = new Schema(optional(1, "id", Types.LongType.get()), optional(2, "name", Types.StringType.get()));
 SparkSession spark = SparkSession.builder().appName("IcebergTest").getOrCreate();
 Catalog catalog = new HiveCatalog(hadoopConfiguration);
 TableIdentifier tableId = TableIdentifier.of("test", "iceberg_table_from_hive_catalog");
 catalog.createTable(tableId, schema, spec);
```
The table created in Hive by the above has DDL like so:
```
CREATE EXTERNAL TABLE `iceberg_table_from_hive_catalog`(
  `id` bigint COMMENT '', 
  `name` string COMMENT '')
ROW FORMAT SERDE 
  'org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe' 
STORED AS INPUTFORMAT 
  'org.apache.hadoop.mapred.FileInputFormat' 
OUTPUTFORMAT 
  'org.apache.hadoop.mapred.FileOutputFormat'
LOCATION
  's3://REDACTED/iceberg_table_from_hive_catalog'
TBLPROPERTIES (
  'metadata_location'='s3://REDACTED/iceberg_table_from_hive_catalog/metadata/00000-7addbbf2-1836-4973-86af-0511ae7577fb.metadata.json', 
  'table_type'='ICEBERG', 
  'transient_lastDdlTime'='1605007216')
```
Which is obviously incorrect as the StorageHandler hasn't been set etc.. I know you worked on a PR that set this all up properly as long as some config/setup was performed at table creation time. Can you please let me know what I need to do and I'll then document it accordingly once I test it working?

Reply:
I think setting `engine.hive.enabled=true` should help.

Reply:
OK, so I tried adding this:
```
hadoopConfiguration.set("engine.hive.enabled", "true");
```
before passing the `hadoopConfiguration` on to the
```
HiveCatalog catalog = new HiveCatalog(hadoopConfiguration);
```
but it doesn't seem to make any difference to the DDL of the table that gets subsequently created.

Reply:
Oh wait, `iceberg.engine.hive.enabled` is what it should be, let me try that...

Reply:
OK, so, with setting that property the DDL looks better:
```
CREATE EXTERNAL TABLE `iceberg_table_from_hive_catalog`(
)
ROW FORMAT SERDE 
  'org.apache.iceberg.mr.hive.HiveIcebergSerDe' 
STORED BY 
  'org.apache.iceberg.mr.hive.HiveIcebergStorageHandler' 

LOCATION
  's3://REDACTED/iceberg_table_from_hive_catalog'
TBLPROPERTIES (
  'metadata_location'='s3://REDACTED/iceberg_table_from_hive_catalog/metadata/00000-95297465-a383-4f0b-b546-dffe7c4ff778.metadata.json', 
  'table_type'='ICEBERG', 
  'transient_lastDdlTime'='1605024789')
```
Not sure whether it's an issue that the schema information is missing? I also can't successfully query this table from Hive so I'll look into that next and see what the issue is. Thanks for helping me over the first hurdle though, I'll update this PR shortly with that required config setting.

Reply:
I got this working when I tested 0.10.0-rc4. In Hive, I had to set `iceberg.mr.catalog=hive` so that the correct catalog was used. Unfortunately, that broke the Hadoop table.

I think we might want to add support for Hadoop tables in our catalogs using an identifier that contains a path to fix this.

Reply:
Also, `engine.hive.enabled` is the table property. When a table has that property, it will be created with the storage handler. I set the property on my table from Spark and it worked.

Reply:
> Also, `engine.hive.enabled` is the table property. When a table has that property, it will be created with the storage handler. I set the property on my table from Spark and it worked.

Really? From what I can see the code uses the value from here https://github.com/apache/iceberg/blob/b4030093b222a96a979d2964eecd6cdd3e9f7608/core/src/main/java/org/apache/iceberg/hadoop/ConfigProperties.java#L27 which is prefixed with `iceberg` as I said above. I tried it both ways and only this way worked but the schema in the Hive table definition is empty so Hive seems to be throwing an error about that when it tries to query it. I haven't tried setting `iceberg.mr.catalog=hive` in Hive when I do the query, I will try that next and see if it makes any difference and what impact that has on the HadoopTables version of it.

Reply:
The logic for setting up the storage handler is here: https://github.com/apache/iceberg/blob/master/hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java#L377-L384

It uses both the environment config and the table property.

Reply:
Yes, exactly, so in this case when I'm creating a new table it doesn't find the value in the table metadata so it drops through to line 383 and looks in the `Configuration` using the `ConfigProperties` key I mentioned above which is `iceberg.engine.hive.enabled`. I stepped through this code to figure out why it wasn't working with `engine.hive.enabled` and saw the property it was looking for in the conf is actually `iceberg.engine.hive.enabled`. So I guess there are two options here:

1. you explicitly create the table with metadata and set `engine.hive.enabled` as a table property.

or 

2. you set the value `iceberg.engine.hive.enabled` explicitly in the conf object (like I'm doing in my example code above) or let this be automatically set by adding it to `hive-site.xml`.

If we agree on this I can document the above here ^. I

I'll then look into what's necessary in order to query the table and add it.

Reply:
CC @lcspinter as he is working on coming up a generic way to handle properties / configuration values wrt Hive.

We would like to come up with solution where the user could easily understand how the different properties are used:
- Iceberg table properties
- HMS table properties
- HMS serde properties
- HiveConf
- Default values

Also we should understand how these properties are used when:
- Reading Hive / Iceberg tables
- Writing Hive / Iceberg tables